packages/console/src/pages/OrganizationRoleDetails/Permissions/use-organization-role-scopes.ts

```diff
@@ -0,0 +1,38 @@
+import { type Scope, type OrganizationScope } from '@logto/schemas';
+import useSWR from 'swr';
+
+import { type RequestError } from '@/hooks/use-api';
+
+/**
+ * Fetches the organization and resource scopes of an organization role.
+ */
+function useOrganizationRoleScopes(organizationRoleId: string) {
+  const organizationRolePath = `api/organization-roles/${organizationRoleId}`;
+
+  const {
+    data: organizationScopes = [],
+    error: fetchOrganizationScopesError,
+    isLoading: isOrganizationScopesLoading,
+    mutate: mutateOrganizationScopes,
+  } = useSWR<OrganizationScope[], RequestError>(`${organizationRolePath}/scopes`);
+
+  const {
+    data: resourceScopes = [],
+    error: fetchResourceScopesError,
+    isLoading: isResourceScopesLoading,
+    mutate: mutateResourceScopes,
+  } = useSWR<Scope[], RequestError>(`${organizationRolePath}/resource-scopes`);
+
+  return {
+    organizationScopes,
+    resourceScopes,
+    error: fetchOrganizationScopesError ?? fetchResourceScopesError,
```

Comment:
Can we return both `fetchOrganizationScopesError` and `fetchResourceScopesError`?

Reply:
We only display one error at a time, so returning both fetchOrganizationScopesError and fetchResourceScopesError separately does not serve a practical purpose. Using this hook treats these two fetch requests as a single entity.